plugins/dev-tools/src/generateFieldTestBlocks.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A field test helper that generates blocks with the field in
+ * various configurations.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+import * as Blockly from 'blockly/core';
+
+
+/**
+ * Generates a number of field testing blocks for a specific field and returns
+ * the toolbox xml string.
+ * @param {string} fieldName The name of the field.
+ * @param {Object|Array.<Object>=} options An options object containing a label
+ *     and an args map that is passed to the field during initialization.  If an
+ *     array is passed, multiple groups of blocks are created each with
+ *     different initialization arguments.
+ * @return {string} The toolbox XML string.
+ */
+export default function generateFieldTestBlocks(fieldName, options) {
+  if (!Array.isArray(options)) {
+    options = [options || {}];
+  }
+
+  let id = 0;
+  let toolboxXml = '';
+  const blocks = [];
+
+  options.forEach((m) => {
+    if (m.label) {
+      // Add label.
+      toolboxXml += `<label text="${m.label}"></label>`;
+    }
+
+    // Define a single field block.
+    const singleFieldBlock = `${++id}test_${fieldName}_single`;
+    blocks.push(
+        {
+          'type': singleFieldBlock,
+          'message0': '%1',
+          'args0': [
+            {
+              'type': fieldName,
+              'name': 'FIELDNAME',
+              ...m.args,
```

Comment:
I'm curious what does  ".." do?

Reply:
This is known as a spread. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Basically it copies over all properties from m.args and adds them to that object.